src/view/LoggedInView.java

```diff
@@ -0,0 +1,152 @@
+
+        JLabel usernameInfo = new JLabel("Currently logged in: ");
+        username = new JLabel();
+
+        JPanel buttons = new JPanel();
+        logout = new JButton(loggedInViewModel.LOGOUT_BUTTON_LABEL);
+        buttons.add(logout);
+
+        requests = new JButton(loggedInViewModel.REQUESTS_BUTTON_LABEL);
+        buttons.add(requests);
+
+        friends = new JButton(loggedInViewModel.FRIENDS_BUTTON_LABEL);
+        buttons.add(friends);
+
+        match = new JButton(loggedInViewModel.MATCH_BUTTON_LABEL);
+        buttons.add(match);
+
+        editProfile = new JButton(loggedInViewModel.EDIT_PROFILE_BUTTON_LABEL);
+        buttons.add(editProfile);
+
+        logout.addActionListener(this);
+        logout.addActionListener(
+            // This creates an anonymous subclass of ActionListener and instantiates it.
+            new ActionListener() {
+                public void actionPerformed(ActionEvent evt) {
+                    if (evt.getSource().equals(logout)) {
+                        logoutController.execute();
+                    }
+                }
+            }
+        );
+
+
+        requests.addActionListener(this);
+        requests.addActionListener(
+            // This creates an anonymous subclass of ActionListener and instantiates it.
+            new ActionListener() {
+                public void actionPerformed(ActionEvent evt) {
+                    if (evt.getSource().equals(requests)) {
+                        displayRequestsController.execute();
+                    }
+                }
+            }
+        );
+
+        friends.addActionListener(this);
+        friends.addActionListener(
+                // This creates an anonymous subclass of ActionListener and instantiates it.
+                new ActionListener() {
+                    public void actionPerformed(ActionEvent evt) {
+                        if (evt.getSource().equals(friends)) {
+                            displayFriendsController.execute();
+                        }
+                    }
+                }
+        );
+
+        match.addActionListener(this);
+        match.addActionListener(
+                // This creates an anonymous subclass of ActionListener and instantiates it.
+                new ActionListener() {
+                    public void actionPerformed(ActionEvent evt) {
+                        if (evt.getSource().equals(match)) {
+                            matchController.execute();
+                        }
+                    }
+                }
+        );
+
+        editProfile.addActionListener(this);
+        editProfile.addActionListener(
+                // This creates an anonymous subclass of ActionListener and instantiates it.
+                new ActionListener() {
+                    public void actionPerformed(ActionEvent evt) {
+                        if (evt.getSource().equals(editProfile)) {
+                            editProfileController.execute();
```

Comment:
We may have to reconsider whether we are having editProfile as a use case itself as it only switches views.

Reply:
I see, so since LoggedIn is also mainly just a view, should we be treating LoggedIn as a use case and have relevant classes like Controller to perform execute? Or do you think something else?

Reply:
I think LoggedIn and EditProfile should mainly be views as the user is not performing actions with them. For instance, when the user arrives at the EditProfile view, they either edit their bio, Spotify handle, or playlist, which are individual use cases.